src/transformers/generation_utils.py

```diff
@@ -386,8 +386,7 @@ def _prepare_decoder_input_ids_for_generation(
     ) -> torch.LongTensor:
         decoder_start_token_id = self._get_decoder_start_token_id(decoder_start_token_id, bos_token_id)
         decoder_input_ids = (
-            torch.ones((input_ids.shape[0], 1), dtype=input_ids.dtype, device=input_ids.device)
-            * decoder_start_token_id
+            torch.ones((input_ids.shape[0], 1), dtype=torch.long, device=input_ids.device) * decoder_start_token_id
```

Comment:
Great change! There is actually no use-case that the decoder_input_ids is not of type torch.long . I like it :-)